src/pages/workerDetails/WorkerDetails.jsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
+
+export const WorkerDetails = () => {
+  const { id } = useParams();
+  const [workerInfoDetails, setWorkerInfoDetails] = useState([]);
+
+  useEffect(() => {
+    fetch(`http://localhost:8000/workers/${id}`)
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        } else {
+          throw new Error("Oops! There is an error, sorry.");
+        }
+      })
+      .then((data) => {
+        setWorkerInfoDetails(data);
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  }, [id]);
```

Comment:
I suggest an empty array is more appropriate, since ID will never change in the lifetime of this component (once arriving at /xxx/:id a brand new component instance is always created)